day-zero/private-plugins.mdx

```diff
@@ -1,5 +1,117 @@
 ---
 title: 'Private Plugins'
-description: 'Explore how platform leads can codify organizational standards using custom plugins in Amplication.'
+description: "Codify organizational standards using Amplication's Private Plugins"
 icon: 'plug'
----
\ No newline at end of file
+---
+
+<Frame>
+    <img src="/images/day-zero/plugins.png" />
+</Frame>
+
+**Plugins** are the driving force behind Amplication's code generation capabilities.
+They provide the flexibility and extensibility needed to adapt Amplication to a wide range of technologies, languages, and frameworks.
+
+When you create a new resource from a Blueprint, it looks at the associated private plugins to understand the concrete steps needed to generate the code and configurations.
+Without plugins, Blueprints would be just abstract descriptions.
+
+<Tip>
+Blueprints define *what* you want to build.
+Plugins define *how* to build it with code.
+</Tip>
+
+## What Are Private Plugins?
+
+Private plugins are **custom code** that hook directly into Amplication's code generation lifecycle events.
+This event system provides plugins with comprehensive access to:
+
+- **Catalog properties** for the specific resource being generated
+- **Blueprint properties** of the specific resource
+- Catalog and blueprint properties of all related resources (set through Blueprint relations)
+
+This deep level of access allows you to create highly customized plugins that can adapt the code generation logic based on your organization's specific needs.
+By combining the Blueprint Plugin events with various property access levels, you can precisely control how your software is generated.
+
+<Note>
+For detailed technical information about implementing plugins, visit our [Plugin Development Guide](/plugins/development).
+</Note>
+
+## How Plugins and Blueprints work together
+
+Blueprints and plugins each serve a distinct purpose.
+Together, they transform high-level specifications into production-ready code.
+
+**Blueprints are declarative**.
+They define the structure and relationships of your software ecosystem.
+For example, a blueprint might specify:
+
+- What entities and fields your service needs
+- How these entities relate to each other
+- What business rules should be enforced
```

Comment:
I dont think we need to talk about entities and fields here... at least not for now (we may get there sometime)
blueprints should describe the architecture components like APIs, workers, services, storage, queues, etc